### PR TITLE
Fix the issue where cluster upgrade doesn't timeout and stuck indefin…

### PR DESCRIPTION
**Pull request by prkalle**
…itely

### What this PR does / why we need it
This PR fixes the issue where the cluster create/upgrade doesn't timeout and stuck indefinitely.
This PR introduces a change where in if the cluster keeps changing the state(current wait logic depends on the cluster state change and would wait if the cluster creation/upgrade is in progress) cli would **also** check if the operation exceeds max timeout (which depends on the timeout value provided by the user). If the operation exceeds maxTimeout it would return error
### Which issue(s) this PR fixes

Fixes # #1230 

### Describe testing done for PR

Create management cluster on Azure and was successfull
Created workload cluster and was success
Upgraded workload cluster and was success

### Release note

```release-note
Fix the issue where where cluster upgrade doesn't timeout and stuck indefinitely
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/tkg/clusterclient/clusterclient.go [hunk 1]
```diff
@@ -486,6 +486,8 @@ func (c *client) WaitForClusterInitialized(clusterName, namespace string) error
 	var lastReason string
 	unchangedCounter := 0
 	interval := 15 * time.Second
+	maxTimeout := 3 * c.operationTimeout
```
Comment (anujc25) [inline pkg/v1/tkg/clusterclient/clusterclient.go:489]: We should add a comment to explain why we added this timeout.

**Review by anujc25 — COMMENTED** [review]
Overall looks good.

File: pkg/v1/tkg/clusterclient/clusterclient.go (hunk 1 above)
Comment (prkalle) [reply pkg/v1/tkg/clusterclient/clusterclient.go:490]: sure, updated

**Review by anujc25 — APPROVED** [review]
lgtm

**Pull request merged**